pkg/apis/hive/v1/agent/ipnet.go

```diff
@@ -0,0 +1,95 @@
+package agent
+
+import (
+	"encoding/json"
+	"net"
+	"reflect"
+
+	"github.com/pkg/errors"
+)
+
+var nullString = "null"
+var nullBytes = []byte(nullString)
+var emptyIPNet = net.IPNet{}
+
+// IPNet wraps net.IPNet to get CIDR serialization.
+type IPNet struct {
+	net.IPNet
+}
```

Comment:
This entire file is copied right out of the installer. Suggestions on how best to handle? We could probably just use the vendored copy we have instead.

Reply:
hmm.. i think we can replace IPNet with just string. For installer this is mainly used to reject install configs with invalid cidr string on unmarshalling, but since this is a CRD this is going to problems in hive code after the object has been accepted.

so the better use here will be to have it as string, and use the validating web hook to enforce the structure.

Reply:
Taking this back to string. Thanks!